### DISCUSSION TITLE
Possible bugs in version 4.1.20-stratego-11

### DISCUSSION THREAD

**Discussion by rgu01** [UPPAALModelChecker/UPPAAL-Meta#174] [Show and tell] [answered]
- When I run UPPAAL 4.1.20-stratego-11 by using “./uppaal” script, the menu shows black blocks as shown in the attached picture. 
- The symbolic simulator does not work if I have hybrid automata with ODE but not instantiating them. In another word, I have ODE in my templates, but I do not instantiate those templates in my model. However, the symbolic simulator refuses to work. 
![Screenshot from 2022-12-27 10-13-44](https://user-images.githubusercontent.com/39125216/209644271-9e22b85d-ab67-4441-9202-2ec983c21532.png)

**Comment by mikucionisaau** [comment, marked as answer]
1. Looks like a bad interaction somewhere between Java and graphics card. I need more information to reproduce it:
 - Have you tried running other Java programs? Any issues there?
 - What Java are you using? Run `java --version` in a terminal to find out.
 - Have you tried a different `Look and Feel` theme? See `Edit` -> `Preferences` -> `Theme`, perhaps the issue is a theme-specific. 
 - What is the Linux distribution? Run `lsb_release -a` in a terminal to find out.
 - What is your Window manager? It looks like Unity from Ubuntu, but I cannot be sure.
 - What is the graphics server? is it Xorg? Wayland? Try `ps aux | grep Xorg` or `ps aux | grep Wayland` to see which one is running.
 - What is the graphics driver? See https://askubuntu.com/a/23240/218500 on how to find out.
 - Are you running in a virtualized environment? Do you have accelerated graphics enabled? 
   Some environments may have experimental 3D acceleration, I know that at least virtualbox may fail to draw some elements when 3D acceleration is enabled, a workaround is to disable it in the virtual machine settings.

2. I have created a feature request: https://github.com/UPPAALModelChecker/UPPAAL-Meta/issues/175

**Reply by rgu01** [reply]
Thank you Marius,

Have you tried running other Java programs? Any issues there?
- I tried my own program with GUI and it works fine.
What Java are you using? Run java --version in a terminal to find out.
- Java 15.0.2
- I am using Oracle VirtualBox and Ubuntu. The 3D acceleration is disabled. Before I continue to answer the rest of your questions, may I mention one thing that may help you find out the issue. I replace the "uppaal" script with the old one that was used in the earlier versions, and it appears to be correct.
[uppaal (old).zip](https://github.com/UPPAALModelChecker/UPPAAL-Meta/files/10308863/uppaal.old.zip)

**Reply by mikucionisaau** [reply]
Right, thanks for the tip! I guess some or all of these JVM options `-Dawt.useSystemAAFontSettings=on -Dswing.aatext=true -Dsun.java2d.opengl=true` trigger something, I suspect that it's probably `opengl`.
Anyways, those options were added to force font-antialiasing when the X server is not configured correctly. I'll remove them and reupload the Linux package, as these options do not appear to be completely benign.

**Reply by mikucionisaau** [reply]
I have updated the zip package, @rgu01 please redownload and test it. It should work.
Thank you for the report!

**Reply by rgu01** [reply]
Thanks! However, which zip package you are referring to? "uppaal (old).zip" that I upload above? Also, it seems the download links for Stratego in https://uppaal.org/downloads/ do not work right now. Once I click one of them, it shows: Forbidden
You don't have permission to access this resource.
 Apache Server at [uppaal.org](mailto:support@its.aau.dk) Port 443. Is it a temporary jitter?

**Reply by mikucionisaau** [reply]
I am sorry, @rgu01 my fault that the permissions got reset. I've fix it now.

**Reply by rgu01** [reply]
Thanks. The new version works fine.
